File: Codes/3396-minimum-number-of-operations-to-make-elements-in-array-distinct.cpp
/*
 * @Author: LetMeFly
 * @Date: 2025-04-08 21:52:04
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-04-08 21:54:03
 */
#if defined(_WIN32) || defined(__APPLE__)
#include "_[1,2]toVector.h"
#endif

class Solution {
public:
    int minimumOperations(vector<int>& nums) {
        unordered_set<int> se;
        for (int i = nums.size() - 1; i >= 0; i--) {
            if (se.count(nums[i])) {
                return min((int)nums.size(), i / 3 + 1);
            }
            se.insert(nums[i]);
        }
        return 0;
    }
};

File: Codes/3396-minimum-number-of-operations-to-make-elements-in-array-distinct.java
/*
 * @Author: LetMeFly
 * @Date: 2025-04-08 21:57:38
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-04-08 21:59:04
 */
import java.util.Set;
import java.util.HashSet;

class Solution {
    public int minimumOperations(int[] nums) {
        Set<Integer> se = new HashSet<>();
        for (int i = nums.length - 1; i >= 0; i--) {
            if (!se.add(nums[i])) {
                return i / 3 + 1;
            }
        }
        return 0;
    }
}